tests/functional/report/monitoring.js

```diff
@@ -0,0 +1,106 @@
+        return new Promise(resolve => http.request({
+            method,
+            host: conf.ipAddress,
+            path,
+            port: 8000,
+            headers: { 'x-scal-report-token': token },
+        }, () => resolve()).end());
+    }
+
+    async function getMetrics() {
+        return new Promise(resolve => {
+            http.get({ host: conf.ipAddress, path: '/metrics', port: 8002 }, res => {
+                assert.strictEqual(res.statusCode, 200);
+
+                let body = '';
+                res.on('data', chunk => { body += chunk; });
+                res.on('end', () => resolve(body));
+            });
+        });
+    }
+
+    function parseMetric(metrics, name, labels) {
+        const labelsString = Object.entries(labels !== undefined ? labels : {
+            method: 'GET', code: '200', route: 'healthcheck',
+        }).map(e => `${e[0]}="${e[1]}"`).join(',');
+        const metric = metrics.match(new RegExp(`^${name}{${labelsString}} (.*)$`, 'm'));
+        return metric ? metric[1] : null;
+    }
+
+    function parseDuration(metrics, labels) {
+        const duration = parseMetric(metrics, 'http_request_duration_seconds_sum', labels);
+        return duration ? parseFloat(duration) : 0;
+    }
+
+    function parseRequestsCount(metrics, labels) {
+        const count = parseMetric(metrics, 'http_requests_total', labels);
+        return count ? parseInt(count, 10) : 0;
+    }
+
+    it('should return system metrics', async () => {
+        const metrics = await getMetrics();
+        assert(metrics.includes('\nnodejs_active_handles_total '));
+    });
+
+    [
+        // Check all methods are reported (on unsupported route)
+        ['/_/fooooo',       { method: 'GET', code: '400' }],
+        ['/_/fooooo',       { method: 'PUT', code: '400' }],
+        ['/_/fooooo',       { method: 'POST', code: '400' }],
+        ['/_/fooooo',       { method: 'DELETE', code: '400' }],
+
+        // S3/api routes
+        ['/',               { method: 'GET', code: '403', route: 'serviceGet' }],
+        ['/foo',            { method: 'GET', code: '404', route: 'bucketGet' }],
+        ['/foo/bar',        { method: 'GET', code: '404', route: 'objectGet' }],
+
+        // Internal handlers
+        ['/_/healthcheck',  { method: 'GET', code: '200', route: 'healthcheck' }],
+        ['/_/healthcheck/deep',
+                            { method: 'GET', code: '200', route: 'deepHealthcheck' }],
+        ['/_/report',       { method: 'GET', code: '200', route: 'report' }],
+        ['/_/backbeat',     { method: 'GET', code: '405', route: 'routeBackbeat' }],
+        ['/_/metadata',     { method: 'GET', code: '403', route: 'routeMetadata' }],
+        ['/_/workflow-engine-operator',
+                            { method: 'GET', code: '405', route: 'routeWorkflowEngineOperator' }],
+    ].forEach(testcase => {
+        const path = testcase[0];
+        const labels = testcase[1];
```

Comment:
Does this work in node 10?
```suggestion
    ].forEach([path, labels] => {
```

Reply:
if you also put parenthesis, it works indeed.
thanks!